db/migrations/00030_hierarchical_groups.sql

```diff
@@ -0,0 +1,22 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE group_hierarchies (
+    id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+    parent_group_id UUID NOT NULL REFERENCES groups(id),
```

Comment:
in the event a group is deleted should the delete cascade ?

Reply:
Yes I think so. I'll need to look at how that's handled in other relations though - I think it might need to be in Go since we emit audit events for things like that...